### DISCUSSION TITLE
Add Support for native .node modules on platforms that include node

### DISCUSSION THREAD

**Discussion by benwis** [remix-run/remix#3201] [Proposals]
I came across this PR for support of .node files, and I'd love to get everybody's opinion on it. It looks like they'd like to do it for remix-tailwind support, but I would like to include generated Node addon modules from napi-rs. 

Personally, I don't see the downside of enabling this behind a config option for people who want it on platforms that support it.
[.Node Pull Request](https://github.com/remix-run/remix/pull/703)

**Comment by benwis** [comment]
I have updated the above pull request patch to work with Remix 1.4.3. I got everything to build, but am having trouble importing the function itself. Trying to learn more about esbuild internals and remix's bundles
[@remix-run+dev+1.4.3.patch.zip](https://github.com/remix-run/remix/files/8703251/%40remix-run%2Bdev%2B1.4.3.patch.zip)

**Reply by benwis** [reply]
Continuing on my Rust compiler adventures, I have the native .node files added to the server build. But I have the most maddening set of errors upon trying to require the file in a loader or action.
```ts
Error: Cannot find module '/build/_assets/napi-rust-functions.darwin-arm64-QSCDSLV7.node'
```
``` 
remix-proj/build
├── _assets
│   └── napi-rust-functions.darwin-arm64-QSCDSLV7.node
├── index.js
└── index.js.map
```
I have created a sandbox with a patch package that should allow you to reproduce the error, and a remix repo fork with the changes thus far. 
https://github.com/benwis/rust-remix
https://github.com/benwis/remix-napi
I'm hoping one of y'all Remix/JS experts can show me what I'm doing wrong. JS imports are hard!

**Reply by benwis** [reply]
It is now working, and I have opened a [pull request ](https://github.com/remix-run/remix/pull/3217) here against the Remix repo for the required changes. The Rust-remix repo mentioned above should also be updated and working. I'll need to clean it up a bit to provide a bit better user experience

**Reply by ragingbit** [reply]
Would love to see this happen (albeit with other packages in mind, such as pretty-bytes).

Any movement on [#3263](https://github.com/remix-run/remix/pull/3263)? I see it hasn't gotten any attention from reviewers for a month and it's a few commits out of date. =)
